kornia/feature/mkd.py

```diff
@@ -0,0 +1,609 @@
+        training_set: (str) Set that model was trained on
+                    'liberty', 'notredame', 'yosemite' ('liberty' is default)
+        reduce_dims: (int) Dimensionality reduction (128 is default)
+    Returns:
+        Tensor: Explicit cartesian or polar embedding
+    Shape:
+        - Input: (B, in_dims, fmap_size, fmap_size)
+        - Output: (B, out_dims, fmap_size, fmap_size)
+    Examples::
+        >>> patches = torch.rand(23, 1, 32, 32)
+        >>> mkd = kornia.feature.mkd.MKD(patch_size=32,
+                                         dtype='concat',
+                                         whitening='pcawt',
+                                         training_set='liberty',
+                                         reduce_dims=128)
+        >>> desc = mkd(patches) # 23x128
+    """
+
+    def __init__(self,
+                 patch_size: int = 32,
+                 dtype: str = 'concat',
+                 whitening: str = 'pcawt',
+                 training_set: str = 'liberty',
+                 reduce_dims: int = 128) -> None:
+        super().__init__()
+
+        self.patch_size = patch_size
+        self.dtype = dtype
+        self.whitening = whitening
+        self.training_set = training_set
+
+        self.sigma = 1.4 * (patch_size / 64)
+        self.smoothing = GaussianBlur2d((5, 5),
+                                        (self.sigma, self.sigma),
+                                        'replicate')
+        self.gradients = MKDGradients()
+
+        # Cartesian embedding with absolute gradients.
+        if dtype in ['cart', 'concat']:
+            ori_abs = EmbedGradients(patch_size=patch_size,
+                                     relative=False)
+            cart_emb = ExplicitSpacialEncoding(dtype='cart',
+                                               fmap_size=patch_size,
+                                               in_dims=ori_abs.kernel.d)
+            self.cart_feats = nn.Sequential(ori_abs, cart_emb)
+
+        # Polar embedding with relative gradients.
+        if dtype in ['polar', 'concat']:
+            ori_rel = EmbedGradients(patch_size=patch_size,
+                                     relative=True)
+            polar_emb = ExplicitSpacialEncoding(dtype='polar',
+                                                fmap_size=patch_size,
+                                                in_dims=ori_rel.kernel.d)
+            self.polar_feats = nn.Sequential(ori_rel, polar_emb)
+
+        self.odims: int = 0
+        if dtype == 'concat':
+            self.odims = polar_emb.odims + cart_emb.odims
+        elif dtype == 'cart':
+            self.odims = cart_emb.odims
+        elif dtype == 'polar':
+            self.odims = polar_emb.odims
+
+        # Compute true reduce_dims.
+        self.reduce_dims: int = min(reduce_dims, self.odims)
+
+        # Load supervised(lw)/unsupervised(pca) model trained on training_set.
+        if self.whitening is not None:
```

Comment:
how can this be None if you define it as a `str` ?

Reply:
Maybe is should use the string 'none' ?